packages/node/src/utils/cosmos.ts

```diff
@@ -64,9 +67,19 @@ export function filterMessages(
   const filters =
     filterOrFilters instanceof Array ? filterOrFilters : [filterOrFilters];
 
-  const filteredMessages = messages.filter((message) => {
-    filters.find((filter) => filterMessageData(message, filter));
-  });
+  const filteredMessages = messages
+    .filter((message) => {
+      filters.find((filter) => filterMessageData(message, filter));
+    })
+    .map((msg) => {
```

Comment:
Would this need to be done if there are no filters too?

Reply:
I just realized that this function will be only be called for custom datasources. It's a seperate mechanism for runtime datasources: https://github.com/subquery/subql-cosmos/blob/3b4468f26316378767954b7b03508c986c3f3ee5/packages/node/src/indexer/indexer.manager.ts#L528

Reply:
And it wouldn't be done if there are no filters but, it has to be done.

Reply:
And this transformation doesn't happen for runtime datasources. `msg.decodedMsg` would still exist.